js/router.js

```diff
@@ -105,10 +114,23 @@ module.exports = Backbone.Router.extend({
   },
 
   newView: function(view, bodyClass, addressBarText){
+    var loadingConfig;
+
     if($('body').attr('id') != bodyClass){
       $('body').attr("id", bodyClass || "");
     }
     $('#obContainer').removeClass("box-borderDashed noScrollBar overflowHidden"); //remove customization styling if present
+    
+    this.pageConnectModal && this.pageConnectModal.remove();
+    this.pageConnectModal = null;
+
+    if (
```

Comment:
I wanted to centralize the Page Connect modal functionality in the router, but since most of the logic to determine whether a page actually connects is in the individual view, it will be up to each page view to opt-in to this functionality. To do so, the view should declare a loadingConfig property that is either an object or a function that returns an object. At the minimum, the object should contain a promise property with a promise that will resolve when the page can load successfully or is rejected when the view determines it can't load the page. Additionally, you can add more properties that specify the messaging that the connecting modal will use. You could see an example of this in the UserPage View down below.